doc/contract-update.md

```diff
@@ -0,0 +1,60 @@
+# Upgrading and Migrating contracts
+
+We consider two approaches:  *Upgrading* existing contracts to a new version but retaining
+state data as-is, and; *migrating* users to a completely new set of contracts.
+
+We would only migrate in the event of a key loss, which prevents upgrading contracts.
+
+## Upgrade contracts approach
+
+Our contracts are already mostly divided into contracts that contain data and contracts that
+contain logic. This is fortuitous for the pattern we are going to use.
+
+`State.sol` contains most of the state, with some in `Shield.sol`.
+This division could be made complete by moving the `proposeBlock` function out of `State.sol`
+and moving the `Shield.sol` data into `State.sol` (there may be a few other minor adjustments
+TBC). `State.sol` will then only contain storage data and getters/setters for the same.
```

Comment:
Setters can also be considered logic and we might want to move this into logic contracts? 
For example, if a straight forward setter is in `State.sol` and a setter function in logic contract calls this `State,sol`, there are instances where this would pass even on a failed execution of setter in `State.sol`. In the event of logic contract pointing to the wrong or destroyed State.sol and if there is no check for done to see if the State contract being exists or has code, then this setting will return true even on a failed execution. One way to go around this, is to check if State.sol exists before calling any setter with in it. But it seems better coverage to have this in logic contract in case there are other considerations we haven't thought of yet.

Reply:
Agree.  This makes sense to me.  I shall remove all setters.

Reply:
Agree.  Let's remove named getters too - it seems silly to have one and not the other.  This means `State.sol` contains no functions (other than build-in getters).  We might need to have a `withdraw` function, only, to facilitate payment.

Reply:
But if the setter is not in `State.sol` how can I set a variable in `State.sol`?